index.js

```diff
@@ -25,56 +28,87 @@ function SonyBraviaTVAccessory(log, config) {
         .on('get', this.getOn.bind(this))
         .on('set', this.setOn.bind(this));
 
+    this.updateTimer();
+}
+
+SonyBraviaTVAccessory.prototype.runTimer = function() {
+    this.getState(function(err, isOn) {
+        if (err == null && isOn != this.isOn) {
+            this.log("State changed: %s", isOn ? "on" : "off");
+            this.service.getCharacteristic(Characteristic.On).updateValue(isOn);
+            this.isOn = isOn;
+        }
+    }.bind(this));
+}
+
+SonyBraviaTVAccessory.prototype.updateTimer = function() {
+    if (this.polling) {
+        clearTimeout(this.timer);
+        setTimeout(function() {
+            this.runTimer();
+            this.updateTimer();
+        }.bind(this), this.interval * 1000);
+    }
+}
+
+SonyBraviaTVAccessory.prototype.getState = function(callback) {
+  var postData = JSON.stringify({
+      method: 'getPowerStatus',
+      params: [],
+      id: 1,
+      version: '1.0'
+  });
+
+  request.post({
+      url: "http://" + this.ipaddress + "/sony/system",
+      headers: {
+          'X-Auth-PSK': this.psk
+      },
+      form: postData
+  }, function(err, response, body) {
+
+      if (!err && response.statusCode == 200) {
+          var json = JSON.parse(body);
+          var status = json.result[0].status;
+          var isOn = status == "active";
+          callback(null, isOn); // success
+      } else {
+          if (response != null) {
+              this.log("Error getting TV status (status code %s): %s", response.statusCode, err);
+              callback(err);
+          } else {
+              callback(null, false);
+          }
+      }
+  }.bind(this));
 }
 
 SonyBraviaTVAccessory.prototype.getOn = function(callback) {
     this.log("Getting whether Sony TV is on...");
 
-    var postData = JSON.stringify({
-        method: 'getPowerStatus',
-        params: [],
-        id: 1,
-        version: '1.0'
-    });
-
-    request.post({
-        url: "http://" + this.ipaddress + "/sony/system",
-        headers: {
-            'X-Auth-PSK': this.psk
-        },
-        form: postData
-    }, function(err, response, body) {
-
-        if (!err && response.statusCode == 200) {
-            var json = JSON.parse(body);
-            var status = json.result[0].status;
-            this.log("TV status is %s", status);
-            var isOn = status == "active";
-            callback(null, isOn); // success
-        } else {
-            if (response != null) {
-                this.log("Error getting TV status (status code %s): %s", response.statusCode, err);
-                callback(err);
-            } else {
-                this.log("Error getting TV status, assuming TV is off: %s.", err);
-                callback(null, false);
-            }
-        }
-    }.bind(this));
+    if (this.polling) {
+        callback(null, this.isOn)
+    } else {
+        this.getState(function(err, isOn) {
+              this.log("State is: %s", isOn ? "on" : "off")
```

Comment:
If getState return an error in the callback, isOn will be undefined.
That will cause an exception in the state logging.
This needs to be fixed so that it won't do the state logging when getState reports an error.